### DISCUSSION TITLE
listing column values according to header

### DISCUSSION THREAD

**Discussion by anjana921** [JoshClose/CsvHelper#1281] [Q&A]
Hello,
I have to list rows for each header from a .csv file and I don't know any information about the data in the file. to do this I first got the headers as a list. And tried to 'getfeild' for each header to store the column values.

e.g 
```
Name |  Subject | Grade
Abby   |  science |  A
Tim     | math     |  B
John   | english   |   C
```
wanted output; 
```
Name
Abby 
Tim
John
```  . i want a list like this for each column.

code :
```cs
List<string> columnValues=new List<string>();
List<string> headers=csv.Context.HeaderRecord.ToList();
foreach(string clm_value in columnValues)
{
While(csv.Read())
{
string FieldValue = csv.GetField<string>(clm_value);
columnValues.Add(FieldValue);
}
}
```

Problem: after it breaks the while loop, it is not doing the Read() for the next header. How do I reopen the read() for next header?

**Comment by JoshClose** [comment]
Can you please ask the question on Stack Overflow? I'm very busy right now and won't be able to answer for a while. https://stackoverflow.com/questions/tagged/csvhelper
